diff --git a/browser/components/moz.build b/browser/components/moz.build
index d1d769ab09..982b571428 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -41,6 +41,7 @@ DIRS += [
     "downloads",
     "enterprisepolicies",
     "extensions",
+    "firefoxcef",
     "firefoxview",
     "genai",
     "ipprotection",
diff --git a/toolkit/library/libxul.symbols b/toolkit/library/libxul.symbols
index e63e3e7442..902d5ee39f 100644
--- a/toolkit/library/libxul.symbols
+++ b/toolkit/library/libxul.symbols
@@ -1,4 +1,17 @@
 XRE_GetBootstrap
+firefox_cef_gecko_set_callbacks
+firefox_cef_gecko_configure
+firefox_cef_gecko_navigate
+firefox_cef_gecko_reload
+firefox_cef_gecko_focus
+firefox_cef_gecko_set_visibility
+firefox_cef_gecko_close
+firefox_cef_gecko_shutdown
+firefox_cef_gecko_post_task
+firefox_cef_gecko_visit_cookies
+firefox_cef_gecko_set_cookie
+firefox_cef_gecko_delete_cookie
+firefox_cef_gecko_run
 uprofiler_get
 #ifdef XP_IOS
 MainProcessInit
diff --git a/browser/moz.configure b/browser/moz.configure
--- a/browser/moz.configure
+++ b/browser/moz.configure
@@ -12,6 +12,7 @@
 
 
 imply_option("MOZ_APP_VENDOR", "Mozilla")
+imply_option("MOZ_APP_UA_NAME", "Firefox")
 imply_option("MOZ_APP_ID", "{ec8030f7-c20a-464f-9b0e-13a3a9e97384}")
 # Include the DevTools client, not just the server (which is the default)
 imply_option("MOZ_DEVTOOLS", "all")
